#! /bin/sh /usr/share/dpatch/dpatch-run
## /tmp/weather.pl-grep-fix.patch.dpatch by  <jari.aalto@cante.net>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: ./Src/weather.pl: Check that /tmp/<station> folder exists before grep

@DPATCH@

diff --git a/Src/weather.pl b/Src/weather.pl
index 6849279..fee14e2 100755
--- a/Src/weather.pl
+++ b/Src/weather.pl
@@ -240,32 +240,37 @@ if($dew eq "")
 }
 
 #check for ok temperature
-open(GREP, "grep Temp $tmpfolder/${station} | ");
-$templine=<GREP>;
-$templine=~m/^Temp:(.*)$/;
+
+if ( -d $tmpfolder/$station )
 {
-	$lattemp=$1;
-	if($temp>-50 && $temp<150)
-	{
-		if($lasttemp>-50 && $lasttemp<150)
-		{
-			if($temp>-2 && $temp<2)
-			{
-				if($lasttemp>7 || $lasttemp<-7)
-				{
-					#probably invalid
-					$station="";
-				}
-			}
-		}
-	}
-	else
-	{
-	#invalid
-	$station="";
-	}
+    open(GREP, "grep Temp $tmpfolder/${station} | ");
+    $templine=<GREP>;
+
+    $templine=~m/^Temp:(.*)$/;
+    {
+	    $lattemp=$1;
+	    if($temp>-50 && $temp<150)
+	    {
+		    if($lasttemp>-50 && $lasttemp<150)
+		    {
+			    if($temp>-2 && $temp<2)
+			    {
+				    if($lasttemp>7 || $lasttemp<-7)
+				    {
+					    #probably invalid
+					    $station="";
+				    }
+			    }
+		    }
+	    }
+	    else
+	    {
+	    #invalid
+	    $station="";
+	    }
+    }
+    close GREP;
 }
-close GREP;
 
 #print "$$station\n";
 if(length($station)==4)
